fdbserver/BlobManifest.actor.cpp

```diff
@@ -191,18 +192,21 @@ class BlobManifest {
 	}
 
 	// Delete all files of oldest manifest
-	static void deleteOldest(const std::vector<BlobManifestFile>& allFiles,
-	                         Reference<BackupContainerFileSystem> container) {
+	ACTOR static Future<Void> deleteOldest(std::vector<BlobManifestFile> allFiles,
+	                                       Reference<BackupContainerFileSystem> container) {
 		if (allFiles.empty()) {
```

Comment:
nit: we can avoid the cost of ACTOR creation/destruction by moving this check at the caller-site, the function can put an assert instead to make sure `allFiles.size() > 0`.

Reply:
the caller already checked that in BlobManifest::count(). if allFiles is empty, it should never call into deleteOldest. So it should not need ACTOR creation/destruction in real prod env.

Reply:
Any reason we aren't making this an `assert` then?

Reply:
although it's not an optimal usage, it seems not like a failure for deleteOldest(). This function tries its best to remove the oldest files if any. Anyway the difference should be subtle to me.. I can convert it to assertion if you insist :)